arangodb-net-standard/IndexApi/IndexApiClient.cs

```diff
@@ -0,0 +1,190 @@
+            {
+                throw new System.ArgumentException("query is required", nameof(query));
+            }
+
+            if (string.IsNullOrEmpty(query.CollectionName))
+            {
+                throw new System.ArgumentException("Collection name is required", nameof(query.CollectionName));
+            }
+
+            uri += '?' + query.ToQueryString();
+            using (var response = await _client.GetAsync(uri).ConfigureAwait(false))
+            {
+                if (response.IsSuccessStatusCode)
+                {
+                    var stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                    return DeserializeJsonFromStream<GetAllCollectionIndexesResponse>(stream);
+                }
+                throw await GetApiErrorException(response).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Creates an index
+        /// </summary>
+        /// <param name="indexType">Type of index to create</param>
+        /// <param name="query">Querystring details</param>
+        /// <param name="body">CreateIndexBody object containing the index details</param>
+        /// <returns></returns>
+        /// <exception cref="System.ArgumentException">Required parameters not provided or invalid.</exception>
+        public virtual async Task<IndexResponseBase> PostIndexAsync(IndexType indexType, PostIndexQuery query, PostIndexBody body)
+        {
+            string uri = _indexApiPath;
+
+            if (query == null)
+            {
+                throw new System.ArgumentException("query is required", nameof(query));
+            }
+
+            if (string.IsNullOrEmpty(query.CollectionName))
+            {
+                throw new System.ArgumentException("Collection name is required", nameof(query.CollectionName));
+            }
+
+            if (body == null)
+            {
+                throw new System.ArgumentException("body is required", nameof(body));
+            }
+
+            switch (indexType)
+            {
+                case IndexType.Persistent:
+                    {
+                        //uri += "#persistent";
+                        body.Type = "persistent";
+                    }
+                    break;
+                case IndexType.FullText:
+                    {
+                        //uri += "#fulltext";
+                        body.Type = "fulltext";
+                    }
+                    break;
+                case IndexType.TTL:
+                    {
+                        //uri += "#ttl";
+                        body.Type = "ttl";
+                    }
+                    break;
+                case IndexType.Geo:
+                    {
+                        //uri += "#geo";
+                        body.Type = "geo";
+                    }
+                    break;
+                default:
+                    throw new System.Exception("Invalid index type");
```

Comment:
This should also be a `System.ArgumentException` because `indexType` is provided as a method argument. It's not possible to encounter this case in normal circumstances though, so not a big deal.